core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -694,7 +694,7 @@ private[storage] class PartiallyUnrolledIterator[T](
   }
 
   override def next(): T = {
-    if (unrolled == null) {
+    if (unrolled == null || unrolled.isEmpty) {
```

Comment:
For consistency with `hasNext`, should this be `!unrolled.hasNext`? Although I don't know this code, this change seems correct. I suppose you could also null out `unrolled` if you find it doesn't have more elements in `hasNext`.